### ISSUE TITLE
example incorrect, clauses not finalized as mandated by FRAT specification

### ISSUE THREAD

**Issue by msoos** [digama0/minisat#1]
Hi,

FRAT specification mandates finalization, see: https://link.springer.com/content/pdf/10.1007%2F978-3-030-72016-2.pdf see: `The clauses may come in any order, but every step that has been added and not deleted must be present.` However, the  minisat that is supposed to provide FRAT proofs, under `https://github.com/digama0/minisat/` doesn't do that. It gives incomplete FRAT proofs, missing the finalization:

```
soos@tiresias:build$ cat a.cnf 
p cnf 2 8
1 2 0
-1 -2 0
1 -2 0
-1 2 0
1 2 0
-1 -2 0
-1 2 0
1 -2 0
soos@tiresias:build$ ./minisat a.cnf b
WARNING: for repeatability, setting FPU to use double precision
============================[ Problem Statistics ]=============================
|                                                                             |
|  Number of variables:             2                                         |
|  Number of clauses:               8                                         |
|  Parse time:                   0.00 s                                       |
|  Simplification time:          0.00 s                                       |
|                                                                             |
===============================================================================
Solved by simplification
restarts              : 0
conflicts             : 0              (0 /sec)
decisions             : 0              (-nan % random) (0 /sec)
propagations          : 1              (541 /sec)
conflict literals     : 0              (-nan % deleted)
Memory used           : 11.00 MB
CPU time              : 0.001848 s

UNSATISFIABLE
soos@tiresias:build$ cat b
o 8  1 2 0
o 16  -1 -2 0
o 24  1 -2 0
o 32  -1 2 0
o 40  1 2 0
o 48  -1 -2 0
o 56  -1 2 0
o 64  1 -2 0
a 1 0
d 24  1 -2 0
a 11  -2 0  l 5 16 0
r
 16 2
 32 8
 48 14
 56 20
 8 26
 40 32
 64 38
 0
soos@tiresias:build$ ~/development/sat_solvers/frat/frat-rs fratchk b
8 orig + 2 added - 1 deleted - 0 finalized = 9
1 missing proofs (50.0%)
type 0: 1
7 original + 9 added never finalized
```

Furthermore, it's missing proofs.

**Comment by digama0** [comment]
This should be opened on the minisat repo, but I can't transfer it for some reason.

**Comment by msoos** [comment]
I wanted to open it there, actually. Unfortunately, the `Issues` tab is not enabled, hence I put it there, since it was the closest I could find :) You probably need to go into the `Settings` of the repo and enable issues?

**Comment by digama0** [comment]
Aha, thanks. FWIW it works correctly if you use `build/debug/bin/minisat_core` instead. I don't actually know what the intention is behind the multiple executables, I will look into this.

**Comment by msoos** [comment]
The other executable is the CDCL+simplifier executable. It does more things than pure CDCL, e.g. it does BVE.

**Comment by digama0** [comment]
The fix in c665adb appears to have also fixed this issue.
